services/QuillLMS/engines/evidence/spec/controllers/evidence/feedback_controller_spec.rb

```diff
@@ -23,6 +24,287 @@ module Evidence
+          expect(parsed_response["optimal"]).to be false
+        end
+      end
+
+      context 'regex2 test' do
+        let!(:rule_regex2) { create(:evidence_regex_rule, :regex_text => "^test", :rule => (rule2)) }
+        let!(:rule2) { create(:evidence_rule, :rule_type => "rules-based-2") }
+        let!(:prompts_rule2) { create(:evidence_prompts_rule, :rule => (rule2), :prompt => (prompt)) }
+
+        it 'should return successfully when there is regex2 feedback' do
+          stub_const("Evidence::Check::ALL_CHECKS", [Check::RegexPostTopic])
+          post :create, params: {entry: "test regex response", prompt_id: prompt.id, session_id: 1, previous_feedback: ([]) }, as: :json
+
+          parsed_response = JSON.parse(response.body)
+          expect(parsed_response["optimal"]).to be false
+        end
+      end
+
+      context 'regex3 test' do
+        let!(:rule_regex3) { create(:evidence_regex_rule, :regex_text => "^test", :rule => (rule3)) }
+        let!(:rule3) { create(:evidence_rule, :rule_type => "rules-based-3") }
+        let!(:prompts_rule3) { create(:evidence_prompts_rule, :rule => (rule3), :prompt => (prompt)) }
+
+        it 'should return successfully when there is regex3 feedback' do
+          stub_const("Evidence::Check::ALL_CHECKS", [Check::RegexTypo])
+          post :create, params: {entry: "test regex response", prompt_id: prompt.id, session_id: 1, previous_feedback: ([]) }, as: :json
+
+          parsed_response = JSON.parse(response.body)
+          expect(parsed_response["optimal"]).to be false
+        end
+      end
+
+      context 'plagiarism test' do
+        it 'should return successfully when there is plagiarism that matches the second plagiarism text string' do
+          stub_const("Evidence::Check::ALL_CHECKS", [Check::Plagiarism])
+          post :create, params: { entry: "bla bla bla #{plagiarized_text2}", prompt_id: prompt.id, session_id: 1, previous_feedback: [] }, as: :json
+
+          parsed_response = JSON.parse(response.body)
+          expect(parsed_response["optimal"]).to be false
+          expect(plagiarized_text2).to(eq(parsed_response["highlight"][0]["text"]))
+          expect(plagiarized_text2).to(eq(parsed_response["highlight"][1]["text"]))
+          expect(first_feedback.text).to(eq(parsed_response["feedback"]))
+          request.env.delete("RAW_POST_DATA")
+          post("plagiarism", :params => ({ :entry => ("bla bla bla #{plagiarized_text2}"), :prompt_id => prompt.id, :session_id => 5, :previous_feedback => ([parsed_response]) }), :as => :json)
+          parsed_response = JSON.parse(response.body)
+          expect(parsed_response["optimal"]).to be false
+          expect(second_feedback.text).to(eq(parsed_response["feedback"]))
+        end
+      end
+
+      context 'prefilter test' do
+        let!(:profanity_rule) do
+          create(:evidence_rule, rule_type: 'prefilter', uid: 'fdee458a-f017-4f9a-a7d4-a72d1143abeb')
+        end
```

Comment:
I don't think we need to specify the `uid` in this `create` call since the factory should provide one if unspecified. 
 Also, pure style thing, but does this need to be a multi-line block?